### DISCUSSION TITLE
Grlb Used On NAJE

### DISCUSSION THREAD

**Discussion by troncatore** [arkypita/LaserGRBL#1220] [General]
Hi ... I am trying in every way known to me to use Grlb laser with the NAJE Master2 but to no avail! I even changed pc! The program is downloaded I also install the driver I restart I open the program all the keys are inactive I connect the machine connects but the program panel remains inactive !!!
Thanks for answering my question
Troncatore

**Comment by gmmanonymus111** [comment]
Might be a silly question, but after successfully connecting to your machine, have you clicked the `padlock` icon to unlock GRBL? By default, GRBL starts in a locked state, to prevent accidental movement/operation.

**Comment by troncatore** [comment]
Sorry no idea about what about you mean! I Just discovered that I have to flash (correct?) the new sistem into the mother board of my NAJE Master2...! I have seen all instruction of It having as a referente Arduino...so i have to do It anyway

**Comment by arkypita** [comment]
I know some people use NEJE with LaserGRBL, so I suppose it's compatible. Maybe it's a problem with your engraver or your computer.

When you press connect, LaserGRBL opens the connection to the device, sends a command asking the machine to send it the current status, then waits for a response from the machine.

Throughout this phase, the state is "connecting".

If the machine does not respond within 10 seconds by sending a correctly formatted status message (i.e. recognizable by LaserGRBL) the connection is closed.

Unfortunately I do not know and do not own the car, so I can only make assumptions.

Make sure there is no need to install any drivers. Does the marker by chance come with a manual? Do they say anything about how to connect it with LaserGRBL?

PS: neje has its own site and provides support, have you tried writing to them? https://www.nejetool.com

----- ITA VERSION -----

So che alcune persone usano NEJE con LaserGRBL, quindi suppongo che sia compatibile. Magari è un problema con la tua marcatrice o con il tuo computer.

Quando premi connect, LaserGRBL apre la connessione verso il dispositivo, manda un comando che chiede alla macchina di inviargli lo stato attuale, quindi resta in attesa di una risposta da parte della macchina.

Durante tutta questa fase lo stato è "connecting".

Se la macchina non risponde entro 10 secondi inviando un messaggio di stato formattato correttamente (ovvero riconoscibile da LaserGRBL) la connessione viene chiusa.

Purtroppo non conosco e non possiedo la macchina, quindi posso solo fare delle ipotesi. 

Verifica che non ci sia bisogno di installare qualche driver. La marcatrice per caso viene fornita con un manuale? Dicono niente su come collegarla con LaserGRBL?

PS: neje ha un suo sito e fornisce supporto, hai provato a scrivere a loro? https://www.nejetool.com

**Reply by troncatore** [reply]
Hi Mate..I discovered that, sorry I'm not into that, I have to install the Grbl 1.1 firmware into the motherboard of my engraver...but I have seen it is a firmware for Arduino...!!! So how I can do that on my engraver?
Thanks a lot

**Reply by arkypita** [reply]
Read and apply instruction provided by neje http://wiki.nejetool.com/doku.php?id=nejelaser_master_2#lasergrbl_software
And see if someone create a video tutorial: https://www.youtube.com/results?search_query=NEJE+grbl

**Comment by arkypita** [comment]
A standard grbl session look like this:
Questo è il carteggio dei messaggi che si scambia LaserGRBL con la macchina:

```
00000	00060836	log	Recording session started @ 18/12/2020 14:44:54
00001	00062415	com	Open COM12 @ 115200 baud DTR, RTS, Ctrl-X
00002	00062418	tx	[18] (machine reset)
00003	00062421	tx	[3F] (request status)
00004	00062622	tx	[3F]
00005	00062823	tx	[3F]
00006	00062961	rx	\r
00007	00062962	rx	Grbl 1.1h ['$' for help]\r (grbl welcome message)
00008	00062970	rx	[MSG:'$H'|'$X' to unlock]\r
00009	00063021	tx	[3F]
00010	00063030	rx	<Alarm|MPos:0.000,0.000,0.000|Bf:15,128|FS:0,0|Pn:S|WCO:0.000,0.000,0.000>\r (grbl status message)
00011	00063222	tx	[3F]
00012	00063230	rx	<Alarm|MPos:0.000,0.000,0.000|Bf:15,128|FS:0,0|Pn:S|Ov:100,100,100>\r
00013	00063423	tx	[3F]
00014	00063430	rx	<Alarm|MPos:0.000,0.000,0.000|Bf:15,128|FS:0,0|Pn:S>\r
00015	00063622	tx	[3F]
00016	00063628	rx	<Alarm|MPos:0.000,0.000,0.000|Bf:15,128|FS:0,0|Pn:S>\r
00017	00063822	tx	[3F]
00018	00063829	rx	<Alarm|MPos:0.000,0.000,0.000|Bf:15,128|FS:0,0|Pn:S>\r
00019	00064023	tx	[3F]
[...]
```

You can record your connect session by hit "? - activate extended log" before hit connect, then upload the content here.
Puoi registrare la tua sessione di connessione cliccando "? - activate extended log" prima di premere "connect", quindi caricare qui il file.

**Comment by troncatore** [comment]
Non ho capito niente ma ci provo...ti farò sapere!!Inviato dal mio Galaxy

**Comment by troncatore** [comment]
This Is the message received during an attempt...i can do It?Best regards....and Happy New year

**Comment by arkypita** [comment]
> This Is the message received during an attempt...i can do It?Best regards....and Happy New 

I did not receive the message. I think that attachment does not work if you reply to github via email
Try by drag-drop here in github page https://github.com/arkypita/LaserGRBL/discussions/1220

**Comment by troncatore** [comment]
This is the message received during at last attempt..I can do it?
![LaserScreen1](https://user-images.githubusercontent.com/76542610/103482927-08c5e000-4de4-11eb-8cd8-7e39e1b37e88.jpg)

**Comment by troncatore** [comment]
naje sent me the correct fles for upgrade to grlb..ALL ok now tanks

**Comment by arkypita** [comment]
Some NEJE machines need to be reprogrammed for use with LaserGRBL. Please follow the directions provided on the [manufacturer's website https://wiki.nejetool.com/doku.php](https://wiki.nejetool.com/doku.php).

**Comment by troncatore** [comment]
Yes..i did io and the engraver are ok...but in the lady version of software.. it's not allowed to put on more buttons...Inviato dal mio Galaxy
-------- Messaggio originale --------Da: arkypita <notifications@github.com> Data: 23/01/21  18:45  (GMT+01:00) A: arkypita/LaserGRBL <LaserGRBL@noreply.github.com> Cc: troncatore <troncatore@yahoo.it>, Author <author@noreply.github.com> Oggetto: Re: [arkypita/LaserGRBL] Grlb Used On NAJE (#1220) 
Some NEJE machines need to be reprogrammed for use with LaserGRBL. Please follow the directions provided on the [https://wiki.nejetool.com/doku.php](manufacturer's website https://wiki.nejetool.com/doku.php).

**Comment by troncatore** [comment]
Finally I discovered how to put on more button!!! Another question is :Use Y axe as rotation AXE..I just discovered that unable the homing due to On Y it is unuseful using a special rotative device. And activating Laser..but It can't works properly, engraving are distorted or speed are too low... What is wrong in my settings?
